File: 06-social-media-scripts/docs/research/ig-tools-bench/reliability.py
"""Fiabilidad: resolver perfil + primer reel en varias cuentas."""
import os, time, json
import instaloader

ACCOUNTS = ["nasa", "openai", "natgeo", "nike", "spotify", "anthropicai", "vercel", "github"]
L = instaloader.Instaloader(quiet=True, save_metadata=False, download_comments=False)
sid = os.environ.get("IG_SESSIONID", "").strip()
if sid:
    L.context._session.cookies.set("sessionid", sid, domain=".instagram.com")
    try:
        L.context.username = L.test_login()
    except Exception as e:
        print("[warn]", e)

out = []
for acc in ACCOUNTS:
    t0 = time.perf_counter()
    try:
        prof = instaloader.Profile.from_username(L.context, acc)
        first = next(iter(prof.get_posts()), None)
        r = {"account": acc, "status": "ok", "t_s": round(time.perf_counter() - t0, 2),
             "followers": prof.followers, "first_post": first.shortcode if first else None}
    except Exception as e:
        r = {"account": acc, "status": "error", "t_s": round(time.perf_counter() - t0, 2),
             "error": type(e).__name__, "msg": str(e)[:110]}
    print(json.dumps(r, ensure_ascii=False))
    out.append(r)
    time.sleep(1.5)

ok = sum(1 for r in out if r["status"] == "ok")
print(f"\ninstaloader fiabilidad: {ok}/{len(out)}")
json.dump({"tool": "instaloader", "results": out}, open("results/reliability_instaloader.json", "w"), indent=1, ensure_ascii=False)

File: 06-social-media-scripts/docs/research/ig-tools-bench/reliability.mjs
/** Fiabilidad: resolver perfil + primer reel en varias cuentas. */
import { igApi } from 'insta-fetcher';
import fs from 'node:fs';

const ACCOUNTS = ['nasa', 'openai', 'natgeo', 'nike', 'spotify', 'anthropicai', 'vercel', 'github'];
const ig = new igApi(`sessionid=${(process.env.IG_SESSIONID ?? '').trim()};`);
const sec = (t) => Number(((performance.now() - t) / 1000).toFixed(2));

const out = [];
for (const acc of ACCOUNTS) {
  const t0 = performance.now();
  let r;
  try {
    const reel = await ig.fetchUserReel(acc, null, 1);
    const m = reel?.xdt_api__v1__clips__user__connection_v2?.edges?.[0]?.node?.media;
    r = { account: acc, status: m ? 'ok' : 'empty', t_s: sec(t0), first_post: m?.code ?? null, play_count: m?.play_count ?? null };
  } catch (e) {
    r = { account: acc, status: 'error', t_s: sec(t0), error: e?.name, msg: String(e?.message).slice(0, 110) };
  }
  console.log(JSON.stringify(r));
  out.push(r);
  await new Promise(x => setTimeout(x, 1500));
}
const ok = out.filter(r => r.status === 'ok').length;
console.log(`\ninsta-fetcher fiabilidad: ${ok}/${out.length}`);
fs.writeFileSync('results/reliability_instafetcher.json', JSON.stringify({ tool: 'insta-fetcher', results: out }, null, 1));
